### PR TITLE
Add support for wallet_getCapabilities v2

### PR DESCRIPTION
**Pull request by lukasrosario**
### _Summary_

**Add filtering support to `wallet_getCapabilities` method**

This PR implements optional chain ID filtering for the `wallet_getCapabilities` RPC method, allowing apps to request capabilities for specific chains only.

**Example Usage:**
```javascript
// Get all capabilities (existing behavior)
await provider.request({
  method: 'wallet_getCapabilities',
  params: ['0x123...']
});

// Get filtered capabilities (new functionality)
await provider.request({
  method: 'wallet_getCapabilities', 
  params: ['0x123...', ['0x1', '0x3']]  // Only return capabilities for chains 1 and 3
});
```

### _How did you test your changes?_

**✅ Comprehensive Test Suite Added:**
- **Parameter validation tests**: 15+ test cases covering all edge cases in `utils.test.ts`
- **Integration tests**: 10+ test cases covering the full request flow in `SCWSigner.test.ts`
- **Error handling tests**: Invalid addresses, unauthorized accounts, malformed parameters
- **Edge case coverage**: Empty filters, undefined capabilities, non-hex keys

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/sign/scw/SCWSigner.test.ts [hunk 1]
```diff
@@ -9,10 +9,10 @@ import { SpendLimit } from ':core/rpc/coinbase_fetchSpendPermissions.js';
...
 import { store } from ':store/store.js';
 import {
-  decryptContent,
-  encryptContent,
-  exportKeyToHexString,
-  importKeyFromHexString,
+    decryptContent,
+    encryptContent,
+    exportKeyToHexString,
+    importKeyFromHexString,
```
Comment (cb-jake) [inline packages/wallet-sdk/src/sign/scw/SCWSigner.test.ts:15]: extra spacing?

**Review by cb-jake — DISMISSED** [review]

File: packages/wallet-sdk/src/sign/scw/SCWSigner.test.ts (hunk 1 above)
Comment (stephancill) [reply packages/wallet-sdk/src/sign/scw/SCWSigner.test.ts:15]: +1

**Review by stephancill — DISMISSED** [review]

**Review by stephancill — APPROVED** [review]

**Review by cb-jake — APPROVED** [review]
